### ISSUE TITLE
Tags/InvalidTypes: report the actual invalid type and tag, not just "at least one tag with an invalid type definition"

### ISSUE THREAD

**Issue by lekemula** [mensfeld/yard-lint#151]
## Description

When `Tags/InvalidTypes` flags a method, the offense message is opaque — it tells you *that* something is wrong but not *which* type or *which* tag, leaving the developer to manually inspect the file and guess which of possibly many tags is the offender.

## Current output

```
$ yard-lint --only Tags/InvalidTypes
Inspecting with 1 validators
Found 10 offense(s):

[W] /Users/.../app/services/brevo_service.rb:22
    InvalidTagType: The `call` has at least one tag with an invalid type definition.
```

## Why this is painful

For a method with several `@param` / `@option` / `@return` tags, each potentially containing multiple type references, this requires reading every tag and cross-referencing against the validator's notion of "valid" to figure out which one to fix. "Invalid" here is a fuzzy notion that depends on what types YARD has loaded — so even careful inspection often doesn't make it obvious which token is being rejected. With 10 such offenses across a repo, triage gets expensive fast.

## What I would expect

Surface the actual invalid type string(s) and (ideally) which tag they came from:

```
[W] /Users/.../app/services/brevo_service.rb:22
    InvalidTagType: `call` has invalid type(s):
      - @return: `SomeUndefinedClass`
      - @param body: `anything`
```

Even just listing the offending type strings without the tag context would already let me fix it in seconds instead of minutes.

## Why this is easy to fix

The validator already computes the list of invalid types — it just throws them away before reporting.

`lib/yard/lint/validators/tags/invalid_types/validator.rb:40-52`:

```ruby
invalid_types = object.docstring.tags
                      .select { |tag| checked_tags.include?(tag.tag_name) }
                      .flat_map(&:types)
                      .compact
                      .uniq
                      .map(&sanitize)
                      .reject { |type| allowed_types.include?(type) }
                      .reject { |type| type_defined?(type) }
                      .reject { |type| type.include?('#') }

return if invalid_types.empty?

collector.puts "#{object.file}:#{object.line}: #{object.title}"
```

`invalid_types` is in scope at the point of `collector.puts` and could be appended to the emitted line, then surfaced by `MessagesBuilder` (`lib/yard/lint/validators/tags/invalid_types/messages_builder.rb`):

```ruby
def call(offense)
  "The `#{offense[:method_name]}` has at least one tag " \
    'with an invalid type definition.'
end
```

Carrying the tag name through would require keeping a `tag -> invalid types` association instead of flattening, but that's still a small change.

A similar improvement applies to other type-validation validators (e.g. `Tags/TypeSyntax`, `Tags/NonAsciiType`) that report a generic "this method has a bad tag" without naming the bad input.

## Bonus: nested `Hash{...}` types and pinpointing the exact failure

A related issue I hit in practice: nested `Hash{K => V}` types (recently added to YARD — see [lsegal/yard#1630](https://github.com/lsegal/yard/pull/1630)) like:

```ruby
@return [Hash{String => Hash{Symbol => Array<String>}}]
```

…appear to be flagged even though the syntax itself is valid YARD now. If yard-lint could (a) understand nested `Hash{}` declarations and (b) on failure report the exact sub-token it rejected (rather than the whole tag), the experience for complex documented return types would be substantially better.

## Version

- yard-lint: **1.5.1**
- Ruby: 3.3.11

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)
